### PR TITLE
Add blst_p2_compress proof

### PR DESCRIPTION
**Pull request by bboston7**

### CODE REVIEW

File: proof/compress-p2.saw [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+let blst_p2_compress_spec is_point_affine = do {
+  out_ptr <- crucible_alloc (llvm_array 96 (llvm_int 8));
+  (inp, inp_ptr) <- ptr_to_fresh_readonly "in" POINTonE2_type;
+  crucible_precond {{ POINTonE2_invariant inp }};
+
+  if is_point_affine then do {
+    // Point is affine, so Z must be 1
+    crucible_precond {{ (fp2_abs (inp.2)) == Fp_2.field_unit }};
+    crucible_precond {{ ~(Fp_2.is_equal (fp2_abs (inp.2), Fp_2.field_zero)) }};
```
Comment (nano-o) [inline proof/compress-p2.saw:174]: Why is this necessary? If Z=1 then it can't be 0.

Comment (bboston7) [reply proof/compress-p2.saw]: I removed this and used a theorem when needed instead.  I also made the same change in the compress_p1 proof.
